src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/DocFile.java

```diff
@@ -202,11 +203,25 @@ public void copyResource(DocPath resource, URL url, boolean replaceNewLine)
      * @throws ResourceIOException if there is a problem while reading the resource
      */
     public void copyResource(DocPath resource, URL url, Resources resources) throws DocFileIOException, ResourceIOException {
-        copyResource(resource, url, true, resources);
+        copyResource(resource, url, resources == null ? UnaryOperator.identity() : line -> localize(line, resources));
     }
 
-    private void copyResource(DocPath resource, URL url, boolean replaceNewLine, Resources resources)
-                throws DocFileIOException, ResourceIOException {
+    /**
+     * Copy the contents of a resource file to this file while transforming and filtering its lines.
+     *
+     * @param resource the path of the resource
+     * @param url the URL of the resource
+     * @param lineTransformer the transforming function that is called for each line; may return
+     * {@code null} to remove a line.
+     *
+     * @throws DocFileIOException if there is a problem while writing the copy
+     * @throws ResourceIOException if there is a problem while reading the resource
+     */
+    public void copyResource(DocPath resource, URL url, UnaryOperator<String> lineTransformer) throws DocFileIOException, ResourceIOException {
```

Comment:
Please wrap the throws clauses on this method and the method below.

Reply:
Done in 6735a2bc2c19e4432008a2e2d0e7a09d132bb9d4